lib/galaxy/tool_util/linters/inputs.py

```diff
@@ -24,6 +24,10 @@ def lint_inputs(tool_xml, lint_ctx):
 
         param_type = param_attrib["type"]
         param_name = param_attrib.get("name", param_attrib.get("argument"))
```

Comment:
You need implement the same as in https://github.com/galaxyproject/galaxy/pull/8409/files#diff-9baf995401cfeb779edf8731ebaf0d2dR256-R260 , otherwise validating a param with no `name` and `argument` that contains `-` will fail.

Reply:
True. I guess the best would be to use the actual function https://github.com/galaxyproject/galaxy/blob/370776bb60a929389667984436698e8a8e8c2ec3/lib/galaxy/tools/parameters/basic.py#L255 instead of reimplementing it here.

Reply:
Yep, maybe extracted from the class.